ui/pages/permissions-connect/snaps/snap-result/snap-result.js

```diff
@@ -34,11 +43,43 @@ export default function SnapResult({
   );
 
   const hasError = !requestState.loading && requestState.error;
-
   const isLoading = requestState.loading;
-
   const snapName = getSnapName(targetSubjectMetadata.origin);
 
+  // Success screen messages
+  let successScreenTitle = t('snapResultSuccess');
+  const successScreenDescription = t('snapResultSuccessDescription', [
+    <Text as={ValidTag.Span} key="1" fontWeight={FontWeight.Medium}>
+      {snapName}
+    </Text>,
+  ]);
+
+  if (requestState.type === 'wallet_installSnap') {
+    successScreenTitle = t('snapInstallSuccess');
+  } else if (requestState.type === 'wallet_updateSnap') {
+    successScreenTitle = t('snapUpdateSuccess');
+  }
+
+  // Failed screen messages
+  let failedScreenTitle = t('snapResultError');
+  let failedScreenDescription;
+
+  if (requestState.type === 'wallet_installSnap') {
+    failedScreenTitle = t('snapInstallationErrorTitle');
+    failedScreenDescription = t('snapInstallationErrorDescription', [
+      <Text as={ValidTag.Span} key="1" fontWeight={FontWeight.Medium}>
+        {snapName}
+      </Text>,
+    ]);
+  } else if (requestState.type === 'wallet_updateSnap') {
+    failedScreenTitle = t('snapUpdateErrorTitle');
+    failedScreenDescription = t('snapUpdateErrorDescription', [
+      <Text as={ValidTag.Span} key="1" fontWeight={FontWeight.Medium}>
+        {snapName}
+      </Text>,
+    ]);
+  }
```

Comment:
I feel like this could be cleaner if it was moved into a separate function or something, which just returns the React component to render.

```ts
const SnapResult = ({ isLoading, hasError }) => {
  function getResult() {
    if (isLoading) {
      return ...; // Loading components
    }

    if (hasError) {
      const title = requestState.type === 'wallet_installSnap' ?
        ? t('snapInstallationErrorTitle')
        : t('snapUpdateErrorDescription');

      // etc.

      return ...; // Error components
    }

    // etc.
  }
};
```

This avoids having a bunch of mutable variables and if-else-statements.

Reply:
There are couple of problems with it. Title and description are different things and each can have different content for update, install and default scenario. The best approach would be to create separate components for these and have them render what's needed, but this ticket was supposed to be only the UI facelift. Also, I think other members are working on some parts of this flow, so I would like to avoid having something like that at this moment. This should be refactored in a different ticket I think.
The usage of nested ternary is disallowed in this repository and ESLint will complain `ESLint: Do not nest ternary expressions.(no-nested-ternary)`. I would like to respect that.

I did some refactoring to reduce complexity around bunch of mutable variables all in one scope and compacted these parts of a template into functions that would make rendering content based on input. It should be more clean now I think.

Reply:
Yeah, I think this needs a refactor, but this seems to be a decently readable. What do you think @Mrtenz ?